src/elements/content-sidebar/BoxAISidebar.tsx

```diff
@@ -1,84 +1,118 @@
-import { Trash } from '@box/blueprint-web-assets/icons/Line';
-import SidebarContent from './SidebarContent';
-import { withAPIContext } from '../common/api-context';
-import { withErrorBoundary } from '../common/error-boundary';
-import { withLogger } from '../common/logger';
-import { ORIGIN_BOXAI_SIDEBAR, SIDEBAR_VIEW_BOXAI } from '../../constants';
-import { EVENT_JS_READY } from '../common/logger/constants';
-import { mark } from '../../utils/performance';
-import { useFeatureEnabled } from '../common/feature-checking';
+import messages from '../common/content-answers/messages';
 
-import messages from '../common/messages';
-import sidebarMessages from './messages';
-import './BoxAISidebar.scss';
+export interface BoxAISidebarContextValues {
+    contentName: string,
+    elementId: string,
+    setCacheValue: (key: 'encodedSession' | 'questions', value: string | null | QuestionType[]) => void,
+    cache: { encodedSession?: string | null, questions?: QuestionType[] },
+    userInfo: { name: string, avatarUrl: string },
+};
 
-const MARK_NAME_JS_READY: string = `${ORIGIN_BOXAI_SIDEBAR}_${EVENT_JS_READY}`;
-
-mark(MARK_NAME_JS_READY);
+export const BoxAISidebarContext = React.createContext<BoxAISidebarContextValues>({
+    setCacheValue: noop,
+    cache: null,
+    contentName: '',
+    elementId: '',
+    userInfo: { name: '', avatarUrl: ''},
+});
 
 export interface BoxAISidebarProps {
-    onClearClick: () => void;
-    agents: AgentType[];
-    selectedAgent: AgentType | null;
+    contentName: string,
+    cache: { encodedSession?: string | null, questions?: QuestionType[] },
+    createSessionRequest: (payload: Record<string, unknown>, itemID: string) => Promise<unknown>,
+    elementId: string,
+    fetchTimeout: Record<string, unknown>;
+    fileExtension: string,
+    fileID: string,
+    getAgentConfig: (payload: Record<string, unknown>) => Promise<unknown>;
+    getAIStudioAgents: () => Promise<unknown>;
+    getAnswer: (payload: Record<string, unknown>,
+        itemID?: string,
+        itemIDs?: Array<string>,
+        state?: Record<string, unknown> ) => Promise<unknown>;
+    getAnswerStreaming: (
+        payload: Record<string, unknown>,
+        itemID?: string,
+        itemIDs?: Array<string>,
+        abortController?: AbortController,
+        state?: Record<string, unknown>,
+    ) => Promise<unknown>,
+    getSuggestedQuestions: (itemID: string) => Promise<unknown> | null;
+    hostAppName: string;
+    isAgentSelectorEnabled: boolean;
+    isAIStudioAgentSelectorEnabled: boolean;
+    isCitationsEnabled: boolean;
+    isDebugModeEnabled: boolean;
+    isIntelligentQueryMode: boolean;
+    isMarkdownEnabled: boolean;
+    isResetChatEnabled: boolean;
+    isStopResponseEnabled: boolean;
+    isStreamingEnabled: boolean;
+    userInfo: { name: '', avatarUrl: ''},
+    setCacheValue: (key: 'encodedSession' | 'questions', value: string | null | QuestionType[]) => void,
```

Comment:
It will also get `recordAction`, it'll be a function that takes an object and return void

Reply:
Added